### ISSUE TITLE
Arm64: Revisit the heuristics for IO completion poller threads

### ISSUE THREAD

**Issue by kunalspathak** [dotnet/runtime#67266]
Today, depending on the number of processors, we create IO completion poller threads. Back then, we did our analysis on older Arm64 machines to come up with the heuristic and they can be different on modern Arm64 machines. We need to revisit them or have some kind of auto-tuning of number of threads creation. 

https://github.com/dotnet/runtime/blob/0e35b8d33851efde52e4e7aaadac239f3c2e97d5/src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEngine.Unix.cs#L52-L57

https://github.com/dotnet/runtime/blob/39b3aace26cc8f27e8d1ea1af0b9d234f7b6d867/src/libraries/System.Private.CoreLib/src/System/Threading/PortableThreadPool.IO.Windows.cs#L41-L43

**Comment by kunalspathak** [comment]
@mangod9

**Comment by mangod9** [comment]
@kouvel as fyi.

**Comment by MihaZupan** [comment]
cc: @dotnet/ncl

**Comment by adamsitnik** [comment]
FWIW I am the person who came up with the magic numbers on Unix. [Here](https://github.com/dotnet/runtime/pull/35800#issuecomment-624719500) you can find the data and reasoning behind it.

Recently after a conversation with @kunalspathak I wanted to give it a quick try and see how the number of epoll threads affects the throughput of JSON Platform benchmark on the Ampere machine. Currently more than 80% of time is spent in `ConcurrentDictionary.TryDequeue` and we need to solve this blocker (#67845, https://github.com/dotnet/aspnetcore/pull/40476) before we try to change the number of epoll threads.

arm64 with 80 cores::

![image](https://user-images.githubusercontent.com/6011991/163327493-ecca8e76-6681-48f1-8b0f-7052173d31cd.png)

x64 with 28 cores:

![image](https://user-images.githubusercontent.com/6011991/163328131-f903636a-cb7e-4120-96a2-4728a5ffba94.png)

**Comment by mangod9** [comment]
yeah we hope to investigate this soon. Might be worth validating again after the concurrent queue fix.

**Comment by kouvel** [comment]
On the Linux Ampere machine, quick tests on JsonPlatform and Json didn't show any significant improvements from decreasing the IO poller thread counts. It still seems like relatively more IO poller threads are necessary on arm64 platforms than on x64 platforms. Probably the heuristics could be more fine-tuned, and I haven't done an exhaustive test.
